core/src/org/javarosa/xform/parse/XFormParser.java

```diff
@@ -967,13 +967,25 @@ protected QuestionDef parseControl(IFormElement parent, Element e, int controlTy
         } else if (ref != null) {
             try {
                 dataRef = new XPathReference(ref);
+                TreeReference controlRefTarget = (TreeReference)dataRef.getReference();
+                if (controlRefTarget.getInstanceName() != null) {
+                    reporter.error("<" + e.getName() +
+                            "> points to an non-main instance (" +
+                            controlRefTarget.getInstanceName() +
+                            "), which isn't supported.");

```

Comment:
warns against something like this
<trigger ref="instance('groups')/root/groups/group">